### PR TITLE
reimplemented the daemon to use inotify instead of watchdog

### LINKED ISSUE #5: watchdog -> inotify?
Recently I had some people complaining about attachments in mails I sent through offlinemsmtp being "broken". After some research, I found the problem: 'larger' attachments were indeed broken, because offlinemsmtp started sending the email before the queue-file was completely written.

https://github.com/sumnerevans/offlinemsmtp/blob/25536f1a59790c6185136e5c9b09bb9b2800e950/offlinemsmtp/daemon.py#L44

was not enough. Changing the `1` to, say, `5` solved my issue, but I obviously don't like that approach. :-)

I found that watchdog doesn't support waiting for the closure of a new file, but inotify does (on Linux, at least).

Is there an interest to replace watchdog with pynotify in offlinemsmtp? If so, I might be interested in taking up the task.

### PR DESCRIPTION
**Pull request by bbbart**
- this allows to listen for queued mail files to be finished writing as not to start sending incomplete messages (resulting in broken attachments, for example)
 - this restricts the daemon to Linux platforms

Closes #5

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,6 +29,7 @@ offlinemsmtp = "offlinemsmtp.__main__:main"
 python = "^3.8"
 PyGObject = "^3.42.0"
 watchdog = "^2.1.6"
```
Comment (sumnerevans) [inline pyproject.toml]: we can get rid of this, right?

File: pyproject.toml [hunk 2]
```diff
@@ -46,6 +47,11 @@ termcolor = "^1.1.0"
 twine = "^3.2.0"
 pre-commit = "^2.17.0"
 
+[tool.poetry.group.dev.dependencies]
+ropevim = "^0.8.1"
+pylint = "^2.15.0"
+pudb = "^2022.1.2"
```
Comment (sumnerevans) [inline pyproject.toml]: Is this some new way of defining dev dependencies?

Comment (bbbart) [reply pyproject.toml]: I'm new to poetry, but I got a warning that `poetry add --dev` was deprecated, and that one should use `poetry add --group=dev` from now on. So I just went with that.

Poetry version 1.2.0

File: pyproject.toml (hunk 1 above)
Comment (bbbart) [reply pyproject.toml]: ah, yes, of course! :-)

Comment (bbbart) [reply pyproject.toml]: (this got fixed in the latest commit in this PR)

**Comment by bbbart** [comment]
I've been using offlinemsmtp with this patch successfully now for a couple of weeks, sending many larg-y attachments (scanned PDFs) without any problems.

Is there an interest in pulling it in, or anything I could do to make it better still?

**Comment by sumnerevans** [comment]
Thanks for the contribution! Sorry it took so long for me to get to it

**Comment by bbbart** [comment]
no problem, glad I could contribute!

**Pull request merged**